### DISCUSSION TITLE
Integrating Passkeys into Login System

### DISCUSSION THREAD

**Discussion by ChocoMeow** [pelican-dev/panel#48] [Ideas]
I've been thinking about how we can improve the login system. One idea I have is to integrate passkeys.

Passkeys are a secure way to authenticate users. They add an extra layer of security and can help prevent unauthorized access. More about passkeys can be found here. https://developers.google.com/identity/passkeys

In simple terms, we would generate a unique passkey for each user during login. This passkey, along with the user's password, would confirm their identity.

**Comment by NaysKutzu** [comment]
I mean sure, but you are just making harder for devs to think at what add-ons they can make for the panel!

**Comment by Miniontoby** [comment]
Why not just add [WebAuthN](https://github.com/pelican-dev/panel/discussions/85), that also adds passkeys but in a simpler way

**Comment by lancepioch** [comment]
Thanks for the input! Passkeys are primarily used to replace passwords, not to be used as 2FA which we already have via OTP.
Additionally the client generates passkeys, not the server.

We believe this would be a great fit for a future plugin!

**Reply by NaysKutzu** [reply]
Thanks good ;)
